### PR TITLE
Added supported for methods other than GET

### PR DESCRIPTION
**Pull request by UnKnoWn-Consortium**
feat: added supported for using method other than GET
feat: method property (if supplied) in option is now honored
refactor: replaced comparators with their strict versions
chore: minor formatting adjustment to increase readability
chore: re-minified the lib

Signed-off-by: UnKnoWn <thomas123456789000@gmail.com>

### CODE REVIEW

**Comment by UnKnoWn-Consortium** [comment]
#148

File: src/eventsource.js [hunk 1]
```diff
@@ -25,20 +25,25 @@
...
-  if (XMLHttpRequest == null) { // https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Using_XMLHttpRequest_in_IE6
+  if (XMLHttpRequest === null) {
+    // https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Using_XMLHttpRequest_in_IE6
     XMLHttpRequest = function () {
       return new ActiveXObject("Microsoft.XMLHTTP");
     };
   }
 
-  if (Object.create == undefined) {
+  if (Object.create === undefined) {
```
Comment (Yaffle) [inline src/eventsource.js:46]: please no

**Comment by Yaffle** [comment]
It is a bit hard to accept the pull request with so many code style changes. I am glad you have looked at the code, but is there any purpose behind them?
The "method" option itself is not supported AFAIK by the native EventSource (and not speced), but it is  a small change, so it is OK to have it.

File: src/eventsource.js (hunk 1 above)
Comment (iMplode-nZ) [reply src/eventsource.js:46]: Object.create can't be null so isn't `===` strictly better to use in this case?

**Comment by iMplode-nZ** [comment]
I'd guess that the reason the code style changed was because he used a formatter.

**Pull request closed without merging**